knowledge-base/grid-state-filter-nullable-bool.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Programatically filter a nullable bool column in the Grid by null values
+description: Learn how to programatically filter nullable bool values
+type: how-to
+page_title: How to filter a Grid column bound to a nullable bool by the null values programatically
+slug: grid-kb-state-filter-nullable-bool
+tags: grid, blazor, column, datagrid, filter, state, nullable, bool
+res_type: kb
+ticketid: 1658561
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+I want to programatically filter a Grid column that is bound to `bool?` by the `null` values. 
+
+
+This KB article also answers the following questions:
+- How to filter a Grid column bound to a `bool?` by the null values programatically
+
+## Solution
+
+To filter a Grid column bound to a `bool?` by the null values programatically:
+
```

Comment:
Adding the relevant steps as a list before the code snippet will improve the KB as this will allow the developer to quickly grasp what are the cornerstones of the proposed implementation.

Using the "see how it's done in the code" approach isn't a particularly good practice.

Reply:
The main reason for the lack of detailed steps is that the client can trigger that in many different ways - on a button click, in the state init event handler, or any other interactive element.

The main part of the KB is to showcase that for null values our clients must use the IsNull filter operator.

Reply:
OK, then mention this milestone explicitly before the code snippet, so that the reader doesn't have to deduce this on their own.

Reply:
Done, you can review the last commit